__tests__/app.test.js

```diff
@@ -205,3 +204,51 @@ describe("PATCH /api/articles/:article_id", () => {
     });
   });
 });
+
+describe("GET /api/articles", () => {
+  describe("api calls", () => {
+    test("200: responds with array of article objects sorted by DESC date with all properties", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body.articles)).toBe(true);
+          expect(body.articles.length).toBe(12);
+          body.articles.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                topic: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                comment_count: expect.any(Number),
+              })
+            );
+          });
+        });
+    });
+    test("200: returns the array of article objects sorted by DESC date", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).not.toBe(0);
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+  });
+  describe("error handling", () => {
+    test("400: Responds with an empty array if the topic query does not exist", () => {
+      return request(app)
+        .get("/api/articles?topic=coconut")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles).toEqual([]);
+        });
+    });
```

Comment:
Would also want a test for if the topic does exist in my database but it just has no articles associated with it - this would be happy path